spec/forms/savings_amounts/savings_amounts_form_spec.rb

```diff
@@ -106,17 +106,26 @@
       end
     end
 
-    context 'check boxes are unchecked' do
-      let(:check_box_params) { check_box_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = '' } }
+    context 'some check boxes are unchecked' do
+      let(:check_box_params) do
+        boxes = check_box_attributes.each_with_object({}) { |attr, hsh| hsh[attr] = '' }
+        boxes[:check_box_isa] = 'true'
+        boxes
+      end
+
+      context 'amounts are valid' do
+        let(:amount_params) { attributes.each_with_object({}) { |attr, hsh| hsh[attr] = Faker::Number.decimal.to_s } }
 
-      shared_examples_for 'it empties amounts' do
         it 'empties amounts' do
           subject.save
           savings_amount.reload
-
           attributes.each do |attr|
             val = savings_amount.send(attr)
-            expect(val).to eq(nil), "Attr #{attr}: expected nil, got #{val}"
+            if attr == :isa
+              expect(val).not_to eq(nil), "Attr #{attr}: expected not to be nil, got #{val}"
+            else
+              expect(val).to eq(nil), "Attr #{attr}: expected nil, got #{val}"
+            end
```

Comment:
Rather than checking the condition in every iteration of the loop, remove `:isa` from the array and test it separately:
```ruby
attributes_except_isa = attributes - [:isa]

attributes_except_isa.each do |attr|
  val = savings_amount.send(attr)
  expect(val).to eq(nil), "Attr #{attr}: expected nil, got #{val}"
end

expect(savings_amount.isa).not_to eq(nil)
```